src/sql/azdata.proposed.d.ts

```diff
@@ -1773,6 +1775,35 @@ declare module 'azdata' {
 		}
 	}
 
+	export namespace metadata {
+		export interface GetServerTableMetadataResult {
+			/**
+			 * The generated create scripts for all tables in every database for a single server.
+			 */
+			scripts: string[];
+		}
+
+		export interface ServerMetadataProvider extends DataProvider {
+			// metadata service methods
+
+			/**
+			 * Generates create scripts for all tables in every database for a single server.
+			 * @param ownerUri Owner URI specifying the server for which metadata will be generated for.
+			 */
+			generateServerTableMetadata(ownerUri: string): Thenable<boolean>;
```

Comment:
Why is this "server table" metadata? Does it really matter what the provider chooses to return from our point of view? If we're going to make this a generic thing for any provider to use then it seems like we should make this as generic as possible. Then it's just up to the provider to determine what it returns that might be useful.